data/Makefile

```
@@ -1,29 +1,27 @@
 # Unicode data generation rules.  Except for the test data files, most
 # users will not use these Makefile rules, which are primarily to re-generate
 # unicode_data.c when we get a new Unicode version or charwidth data; they
-# require ruby and julia to be installed.
+# require julia to be installed.
 
 # programs
 CURL=curl
-RUBY=ruby
 PERL=perl
 MAKE=make
 JULIA=julia
 CURLFLAGS = --retry 5 --location
 
-.PHONY: clean
+.PHONY: clean rawdata
 
 .DELETE_ON_ERROR:
 
-utf8proc_data.c.new: data_generator.rb UnicodeData.txt GraphemeBreakProperty.txt DerivedCoreProperties.txt CompositionExclusions.txt CaseFolding.txt CharWidths.txt emoji-data.txt
-	$(RUBY) data_generator.rb < UnicodeData.txt > $@
-
-CharWidths.txt: charwidths.jl EastAsianWidth.txt
-	$(JULIA) charwidths.jl > $@
+utf8proc_data.c.new: data_generator.jl rawdata
+	$(JULIA) --project=. data_generator.jl > $@
```

We may need to add a new target to initialize the julia environment first.

```suggestion
utf8proc_data.c.new: data_generator.jl rawdata julia-pkg-instantiated
	$(JULIA) --project=. data_generator.jl > $@

julia-pkg-instantiated:
	$(JULIA) --project=. -e 'using Pkg; Pkg.instantiate()'
	echo 1 > $@
```

---

Since we only need it for the generator, maybe just combine the rules:
```suggestion
utf8proc_data.c.new: data_generator.jl rawdata
	$(JULIA) --project=. -e 'using Pkg; Pkg.instantiate()'
	$(JULIA) --project=. data_generator.jl > $@
```